### PR TITLE
Add support for button styling for ECE buttons

### LINKED ISSUE #8951: Ensure Button Settings are applied in ECE and Obtain Design Approval
Ensure that the settings for express checkout buttons in the WooPayments settings page are correctly applied to the Express Checkout Element (ECE) buttons. Additionally, secure final design approval for ECE buttons in both shortcode and blocks checkout.
```[tasklist]
### Tasks
- [x] Settings from WooPayments intended for PRBs are now appropriately applied to ECE buttons.
- [x] See if we can default to showing both GPay and APay together when available instead of showing a `See more` link to show both buttons.
- [x] Design Review.
- [x] Incorporate any feedback from the review and finalize the design.
```

### PR DESCRIPTION
**Pull request by reykjalin**
Fixes #8951 

## Changes proposed in this Pull Request

- Respect Apple Pay and Google Pay button settings when using ECE buttons.

No automated tests because it's hard to make assertions on iframes and their style.

## Testing instructions

> [!IMPORTANT]
> There's an issue with Storefront that causes the button width to be incorrect. We're still discussing how to address that. In the meantime you can test with Twenty-Twenty Four. See https://github.com/Automattic/woocommerce-payments/issues/8951#issuecomment-2183378659 for details.

> [!IMPORTANT]
> HTTPS is a **hard requirement** for testing the buttons. They will not be rendered unless you have a valid HTTPS URL to test with. Jurassic tube is recommended for permanent custom URLs.

### Setup

1. Make sure you have a valid HTTPS URL you can use for testing the buttons.
2. Register the domain for Apple Pay with the following CLI command:

```bash
curl https://api.stripe.com/v1/payment_method_domains \
     # Don't forget the `:` at the end!
  -u "<secret key, sk_xxx or rk_xxx>:" \
  -H "Stripe-Account: <merchant account ID>" \
     # Example: my.domain.com
  -d domain_name="<your_url>"
```

3. Enable ECE buttons by setting the `_wcpay_wcpay_feature_stripe_ece` to `1`: `wp option set _wcpay_feature_stripe_ece 1`.
4. Keep a tab open on **wp-admin &rarr; Payments &rarr; Settings &rarr; Express checkouts &rarr; Apple Pay/Google Pay &rarr; Customize**. You'll be making changes there during the tests.
5. Add a simple product to your cart.

### Test button sizes

> [!NOTE]
> Due to limitations in Stripe's API the maximum height of the buttons is `55px` instead of the `56px` suggested by the `Large` option in the settings. We're discussing this in https://github.com/Automattic/woocommerce-payments/issues/8951#issuecomment-2183177946.

1. Select a size in the settings.
2. Open the shortcode Cart page.
3. Inspect the button using your browser's developer tools.
4. Make sure the height matches the setting.
5. Repeat steps (2)-(4) for the shortcode Checkout, and Blocks Cart and Checkout pages.
6. Repeat steps (1)-(5) for the other 2 sizes.

### Test button theme (i.e. color)

> [!IMPORTANT]
> Stripe only supports an `Outline` version of the Google Pay button so both the `Light` and `Outline` themes will have an outline.

> [!NOTE]
> You can keep the size the same while doing this test.

1. Select a theme in the settings.
2. Open the shortcode Cart page.
3. Make sure the button theme matches the setting.
5. Repeat steps (2)-(4) for the shortcode Checkout, and Blocks Cart and Checkout pages.
6. Repeat steps (1)-(5) for the other 2 sizes.

### Test combinations

- Test other combinations of themes and sizes (not necessarily exhaustive, pick 2-3 additional combinations you haven't tested) and make sure the settings are applied correctly.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _QA Testing Not Applicable_
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable. : _Not Applicable_
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. _Not Applicable_

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8988` or branch name `add/8951-support-style-settings-with-ece-buttons` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/8951-support-style-settings-with-ece-buttons) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 3a3dd6c0722b408085a3bd5055617abcb0cf54d6
- Build time: 2024-06-25 04:00:44 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by reykjalin** [comment]
@nikkivias @elizaan36 the screenshots of the different options are in https://github.com/Automattic/woocommerce-payments/issues/8951#issuecomment-2179646209

**Review by elizaan36 — APPROVED** [review]
LGTM! Thanks!

**Review by rafaelzaleski — COMMENTED** [review]
Edit: Just saw this in the PR description 😅 

The light version of the button is the same as the outline version. Also, I believe we need to review the font sizes, as there is a noticeable difference between the WooPay button and the ECE button.

Screen Shot 2024-06-24 at 16 41 19 PM

**Comment by reykjalin** [comment]
> Also, I believe we need to review the font sizes, as there is a noticeable difference between the WooPay button and the ECE button.

We don't have _direct_ control over font sizes, but we can set the [base font size](https://docs.stripe.com/elements/appearance-api?platform=web#commonly-used-variables) for Stripe's elements, so we can probably figure out a way to get the exact font size we want.

I'm curious if this should be a separate issue though? Are the font size not different in the PRB implementation? 🤔 If they're different there I'd say this isn't a regression and shouldn't be fixed in this PR.

File: client/express-checkout/utils/index.ts [hunk 1]
```diff
@@ -108,3 +110,68 @@ export const getErrorMessageFromNotice = ( notice: string ) => {
...
+			),
+			applePay: mapWooPaymentsThemeToButtonTheme(
+				'applePay',
+				buttonSettings?.theme ?? 'black'
+			),
+		},
+		buttonType: {
+			googlePay: buttonSettings?.type ?? 'buy',
+			applePay: buttonSettings?.type ?? 'plain',
+		},
```
Comment (rafaelzaleski) [inline client/express-checkout/utils/index.ts]: When I select the call to action `only icon`, nothing is rendered. In this case the value of `buttonSettings.type` is `default`.

**Review by rafaelzaleski — CHANGES_REQUESTED** [review]

File: client/express-checkout/utils/index.ts (hunk 1 above)
Comment (rafaelzaleski) [reply client/express-checkout/utils/index.ts]:
``` 
Uncaught IntegrationError: Invalid value for elements.create('expressCheckout', options): buttonType.applePay should be one of the following strings: add-money, book, buy, check-out, contribute, donate, order, plain, reload, rent, subscribe, support, tip, top-up. You specified: default.
```

Comment (reykjalin) [reply client/express-checkout/utils/index.ts]: Nice catch, this should be a _relatively_ simple fix. Thank you!

**Comment by rafaelzaleski** [comment]
> We don't have direct control over font sizes, but we can set the [base font size](https://docs.stripe.com/elements/appearance-api?platform=web#commonly-used-variables) for Stripe's elements, so we can probably figure out a way to get the exact font size we want.

> I'm curious if this should be a separate issue though? Are the font size not different in the PRB implementation? 🤔 If they're different there I'd say this isn't a regression and shouldn't be fixed in this PR.

I think it fits the scope of https://github.com/Automattic/woocommerce-payments/issues/8136.
I noticed that the GPay button uses an svg for the text and I think we should probably do the same in WooPay: https://github.com/Automattic/woocommerce-payments/issues/8136#issuecomment-2187356240

**Comment by reykjalin** [comment]
> I think it fits the scope of https://github.com/Automattic/woocommerce-payments/issues/8136.

Ah yes, I agree!

> I noticed that the GPay button uses an svg for the text and I think we should probably do the same in WooPay

Yes, that sounds like a nice solution. It'd ensure it scales with the size of the button effortlessly too 👍

File: client/express-checkout/utils/index.ts (hunk 1 above)
Comment (reykjalin) [reply client/express-checkout/utils/index.ts]: Fixed in 3a3dd6c0722b408085a3bd5055617abcb0cf54d6 🙌

**Review by rafaelzaleski — APPROVED** [review]
The code looks good, and the settings are correctly applied to the ECE buttons. Nice work!

There are some design inconsistencies, but they are out of scope for this review and have already been addressed in issue #8951 with the design team, resulting in new issues being created.

**Pull request merged**